scripts/deploy-federation.sh

```diff
@@ -122,14 +122,14 @@ if [[ "${NAMESPACED}" ]]; then
   kubectl -n "${NS}" create rolebinding federation-admin --clusterrole=cluster-admin --serviceaccount="${NS}:default"
 else
   # TODO(marun) Make this more restrictive.
-  kubectl create clusterrolebinding federation-admin --clusterrole=cluster-admin --serviceaccount="${NS}:default"
+  kubectl create clusterrolebinding federation-admin-${NS} --clusterrole=cluster-admin --serviceaccount="${NS}:default"
 fi
 
 if [[ ! "${USE_LATEST}" ]]; then
   if [[ "${NAMESPACED}" ]]; then
     INSTALL_YAML="${INSTALL_YAML}" IMAGE_NAME="${IMAGE_NAME}" scripts/generate-namespaced-install-yaml.sh
   else
-    INSTALL_YAML="${INSTALL_YAML}" IMAGE_NAME="${IMAGE_NAME}" scripts/generate-install-yaml.sh
+    INSTALL_YAML="${INSTALL_YAML}" IMAGE_NAME="${IMAGE_NAME}" FEDERATION_NAMESPACE="${NS}" NAMESPACED="${NAMESPACED}" scripts/generate-install-yaml.sh
```

Comment:
This command is already `!${NAMESPACED}`, the reason I need this is because the script `scripts/generate-namespaced-install-yaml.sh` also need to call `scripts/generate-install-yaml.sh`, so in `scripts/generate-install-yaml.sh`, I need to make sure the `--federation-namespace` was only added when `${FEDERATION_NAMESPACE}` set and  `!${NAMESPACED}`, make sense?

Reply:
Since the namespaced install does not set an explicit namespace (it is intended to be applied with `-n <namespace>`), the namespaced generation script will never supply `FEDERATION_NAMESPACE` to this script.  That suggests that supplying `NAMESPACED` is redundant.

Reply:
Yes, the namespaced install yaml does not set an explicit namespace (it is intended to be applied with `-n <namespace>`), but the `-n <namespace>` was set by `FEDERATION_NAMESPACE` in `deploy-federation.sh`, am I missing anything? https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/userguide.md#automated-deployment-1

Reply:
The script I was referring to was `scripts/generate-install-yaml.sh` not `scripts/deploy-federation.sh`.  I'm suggesting that `scripts/generate-install-yaml.sh` should not accept `NAMESPACED`, and assume that if `FEDERATION_NAMESPACE` is supplied, that the configuration is not for a namespaced deployment.

Reply:
If I remove `NAMESPACED` from `scripts/generate-install-yaml.sh`, then the following command will generate two `- --federation-namespace='${FEDERATION_NAMESPACE}'`.

```
NAMESPACED=y FEDERATION_NAMESPACE=<namespace> scripts/deploy-federation.sh <image name>
```

As `scripts/generate-namespaced-install-yaml.sh` also need to call `scripts/generate-install-yaml.sh`, if I do not check `NAMESPACED` in `scripts/generate-install-yaml.sh`, it will generate two `- --federation-namespace='${FEDERATION_NAMESPACE}'` in federation controller statefulset.

Reply:
When I apply [this diff](https://gist.github.com/marun/333e4b640f8f35adeaa4e4e14e03bb24), the generated yaml does not does not contain duplicate `--federation-namespace` arguments.  Am I missing something?

Reply:
This is exactly the same diff that I want you test. Please run the command as follows, you will see there are two `- --federation-namespace=`.

1.
```
NAMESPACED=y FEDERATION_NAMESPACE=test-namespace scripts/deploy-federation.sh  gyliu/federation-v2:canary cluster2
```
2. 
After cluster2 and cluster1 joined, run 
```
kubectl edit sts -n test-namespace
```

You will see follows:
```
  spec:
      containers:
      - args:
        - --install-crds=false
        - --limited-scope=true
        - --federation-namespace=$(FEDERATION_NAMESPACE) <<<<<<
        - --registry-namespace=$(CLUSTER_REGISTRY_NAMESPACE)
        - --federation-namespace=test-namespace <<<<<<
        command:
```

Reply:
Ah, I was testing the generation separate from the deploy script.  I forgot that environment variables passed on the command line are inherited by child processes, so that the `FEDERATION_NAMESPACE` passed to the deploy script would end up in the yaml generation script 3 calls deep.  I would rather see the call in the namespaced yaml generation script [clear the env variable](https://gist.github.com/marun/333e4b640f8f35adeaa4e4e14e03bb24#file-ns-diff-L44) rather than adding the `NAMESPACE` control variable.

Reply:
@marun thanks, updated.